michelve/software-license-manager
# 6.2 dev

Description
This pull request introduces significant improvements and features for the SLM Plus plugin, including the addition of custom blocks, a new block category, enhanced shortcodes, and customizable hooks. These updates aim to enhance the usability, flexibility, and customization of the plugin for developers and end-users.

Key Features and Changes
Custom Block Category:

Added a new block category, "SLM Plus", to organize related blocks under one section in the WordPress block editor.
Ensures a clean and intuitive block inserter experience.
New Blocks:

Forgot License Block:
Displays a form where users can retrieve license information by entering their email.
Provides an interactive preview in the block editor.
List Licenses Block:
Dynamically displays a list of licenses associated with the logged-in user.
Includes a placeholder preview in the editor for better visualization.
Improved Shortcodes:

[slm_forgot_license]: Displays the "Forgot License" form for users to retrieve their license information.
[slm_list_licenses]: Displays a table of licenses dynamically based on the logged-in user's email.
Custom Hooks for Flexibility:

Added the following hooks to allow developers to customize text and logic:
slm_invalid_email_message: Customize the invalid email error message.
slm_success_message: Customize the success message.
slm_no_license_message: Customize the "no license found" message.
slm_license_email_message: Modify the license email content.
slm_license_email_subject: Modify the license email subject.
slm_form_label: Customize the form label text.
slm_form_button_text: Customize the form button text.
Editor Enhancements:

Blocks now have unique previews in the block editor:
The "Forgot License" block shows a form preview.
The "List Licenses" block shows a sample list of licenses for better context.
Styling Improvements:

Added slm-blocks.css for consistent and responsive styling across blocks and shortcodes.
Includes clean, accessible table styles for the "List Licenses" block.
How to Use
Blocks:

Forgot License Block: Add it to a page or post from the "SLM Plus" category in the block editor.
List Licenses Block: Add it from the same category to display user licenses dynamically.
Shortcodes:

Use [slm_forgot_license] to render the Forgot License form.
Use [slm_list_licenses] to display a license table for logged-in users.
Customization:

Apply the new hooks in your theme or custom plugin to personalize messages, email content, and form labels.
Testing Steps
Custom Category:

Verify the new "SLM Plus" category appears in the block editor.
Ensure both blocks are listed under this category.
Blocks:

Add both blocks to a post or page and confirm the preview and frontend rendering.
Shortcodes:

Test the [slm_forgot_license] and [slm_list_licenses] shortcodes on various pages with different user roles.
Custom Hooks:

Add custom filters for each hook and verify changes in the form, messages, and emails.
Styling:

Confirm consistent styling for both block previews and frontend output.
Impact
Enhances plugin usability by organizing blocks and providing visual previews in the editor.
Improves developer experience with hooks and shortcodes for seamless customization.
Increases flexibility for integrating SLM Plus into a variety of projects.
This pull request is a comprehensive update that improves functionality, user experience, and extensibility for the SLM Plus plugin.
